# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/stf/squeak/squeak-head-tail.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -dur Squeak-3.6-3/platforms/unix/config/inisqueak.in Squeak-3.6-3-patch/platforms/unix/config/inisqueak.in
--- Squeak-3.6-3/platforms/unix/config/inisqueak.in	2003-08-31 07:31:35.000000000 +0200
+++ Squeak-3.6-3-patch/platforms/unix/config/inisqueak.in	2004-09-26 13:12:57.028629888 +0200
@@ -191,7 +191,7 @@
       echo
     fi
   done
-  IMAGE=`echo ${images} | tr ' ' '\012' | tail +${reply} | head -1`
+  IMAGE=`echo ${images} | tr ' ' '\012' | tail -n+${reply} | head -n1`
   CHANGES=${IMAGE}.changes.gz
   IMAGE=${IMAGE}.image.gz
 fi
diff -dur Squeak-3.6-3/platforms/unix/config/mkconfig.in Squeak-3.6-3-patch/platforms/unix/config/mkconfig.in
--- Squeak-3.6-3/platforms/unix/config/mkconfig.in	2003-02-10 04:28:32.000000000 +0100
+++ Squeak-3.6-3-patch/platforms/unix/config/mkconfig.in	2004-09-26 13:13:30.157593520 +0200
@@ -2,8 +2,8 @@
 
 int_modules="@int_modules@"
 ext_modules="@ext_modules@"
-int_plugins="@int_plugins@ `tail -1 @vmmdir@/plugins.int | sed 's,.* =,,'`"
-ext_plugins="@ext_plugins@ `tail -1 @vmmdir@/plugins.ext | sed 's,.* =,,'`"
+int_plugins="@int_plugins@ `tail -n1 @vmmdir@/plugins.int | sed 's,.* =,,'`"
+ext_plugins="@ext_plugins@ `tail -n1 @vmmdir@/plugins.ext | sed 's,.* =,,'`"
 
 if test -f plugins.exc; then
   nlsp='tr \012\015 \040\040'
--- ./platforms/unix/config/verstamp	2002-10-27 02:29:04.000000000 +0200
+++ ./platforms/unix/config/verstamp-p	2004-11-03 02:52:26.487000096 +0100
@@ -5,14 +5,14 @@
 # Last edited: 2002-06-08 18:32:53 by piumarta on emilia.inria.fr
 
 if test -f $1; then
-  SERIAL=`head -1 $1 | cut -d ' ' -f 3 | sed 's/;//'`
+  SERIAL=`head -n1 $1 | cut -d ' ' -f 3 | sed 's/;//'`
   SERIAL=`expr $SERIAL + 1`
 else
   SERIAL=1
 fi
 
 if expr "$2" : ".*gcc" >/dev/null ; then
-  CC=`$2 -v 2>&1 | tail -1 | sed 's/.*based on//g' | awk '{print $1 , $3} '`
+  CC=`$2 -v 2>&1 | tail -n1 | sed 's/.*based on//g' | awk '{print $1 , $3} '`
 else
   CC=$2
 fi
--- ./platforms/unix/config/ltcf-c.sh	2002-11-27 15:21:02.000000000 +0100
+++ ./platforms/unix/config/ltcf-c.sh-p	2004-11-03 02:54:22.545356544 +0100
@@ -151,7 +151,7 @@
     # If the export-symbols file already is a .def file (1st line
     # is EXPORTS), use it as is.
     # If DATA tags from a recent dlltool are present, honour them!
-    archive_expsym_cmds='if test "x`head -1 $export_symbols`" = xEXPORTS; then
+    archive_expsym_cmds='if test "x`head -n1 $export_symbols`" = xEXPORTS; then
         cp $export_symbols $output_objdir/$soname-def;
       else
         echo EXPORTS > $output_objdir/$soname-def;
